src/client/dfuse/SConscript

```diff
@@ -101,7 +105,10 @@ def build_client_libs_shared(env, prereqs):
     pil4dfs_lib = pil4dfsenv.d_library('pil4dfs/libpil4dfs', pil4dfs_obj)
     env.InstallVersionedLib(os.path.join("$PREFIX", 'lib64'), pil4dfs_lib)
 
-    gen_script = ilenv.d_program('il/gen_script', ['il/gen_script.c'], LIBS=[])
+    libs = []
+    if GetOption("test_coverage"):
```

Comment:
Just a thought.  Rather than sprinkling these checks everywhere, why not just do something in the various builders

https://github.com/daos-stack/daos/blob/367d2667a213b50ddda8ae479580f417bc5adf96/site_scons/site_tools/daos_builder.py#L167

We call d_program, d_test_program, d_library, etc all over the place.  If you just put an Append inside those routines, you can limit this to a few LOC

Those functions are added here: https://github.com/daos-stack/daos/blob/367d2667a213b50ddda8ae479580f417bc5adf96/site_scons/site_tools/daos_builder.py#L251-L261